deps/secure-socket/biowrap.lua

```diff
@@ -48,6 +48,10 @@ return function (ctx, isServer, socket, handshakeComplete, servername)
 
   local function handshake(callback)
     if ssl:handshake() then
+      local cert = ssl:peer()
+      if not cert:check_host(servername) then
+        return error("The server hostname does not match the certificate's domain")
```

Comment:
Is there a reason this throws an error instead of pushing the error to the `handshakeComplete` callback like peer verification below?